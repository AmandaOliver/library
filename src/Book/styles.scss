.book {
    width: 100%;
    height: 50px;
    background-color: #e6e6e6;
    margin: 5px;
    transition: box-shadow 0.2s ease-in-out 0s;
    font-family: Georgia, 'Times New Roman', Times, serif;
    display: flex;
    flex-flow: wrap row;
    justify-content: space-around;
    align-items: center;
    &__property {
        width: 200px;
        &__name{
            color: #3fbfbb;
            font-size: 16px;
        }
    }
}
