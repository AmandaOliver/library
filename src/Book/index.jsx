import React from 'react'
import moment from 'moment'
import './styles.scss'


const Book = ({ book, author }) => {

  const { name, genre, publishDate } = book
  const publishDateFormatted = moment(publishDate).format('DD/MM/YYYY')

  return (
    <div className="book">
      <div className='book__property'>
        <span className="book__property__name">Name: </span>
        <span className="book__property__value">{name} </span>
      </div>
      <div className='book__property'>
        <span className="book__property__name">Genre: </span>
        <span className="book__property__value">{genre} </span>
      </div>
      <div className='book__property'>
        <span className="book__property__name">Publish Date: </span>
        <span className="book__property__value">{publishDateFormatted}</span>
      </div>
      <div className='book__property'>
        <span className="book__property__name">Author Name: </span>
        <span className="book__property__value">{author.name} </span>
      </div>
      <div className='book__property'>
        <span className="book__property__name">Author Gender: </span>
        <span className="book__property__value">{author.gender} </span>
      </div>
    </div>
  )
}

export default Book;
