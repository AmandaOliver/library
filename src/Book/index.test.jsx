import React from 'react'
import renderer from 'react-test-renderer';
import Book from '.'

const book = {
    name: 'mockName',
    genre: 'mockGenre',
    publishDate: new Date(),
}

const author = {
    name: 'mockName',
    gender: 'mockGenre'
}

describe('Book', () => {
  it('renders', () => {
    const component = renderer.create(
        <Book book={book} author={author}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
