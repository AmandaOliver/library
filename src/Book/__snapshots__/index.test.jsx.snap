// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders 1`] = `
<div
  className="book"
>
  <div
    className="book__property"
  >
    <span
      className="book__property__name"
    >
      Name: 
    </span>
    <span
      className="book__property__value"
    >
      mockName
       
    </span>
  </div>
  <div
    className="book__property"
  >
    <span
      className="book__property__name"
    >
      Genre: 
    </span>
    <span
      className="book__property__value"
    >
      mockGenre
       
    </span>
  </div>
  <div
    className="book__property"
  >
    <span
      className="book__property__name"
    >
      Publish Date: 
    </span>
    <span
      className="book__property__value"
    >
      02/02/2019
    </span>
  </div>
  <div
    className="book__property"
  >
    <span
      className="book__property__name"
    >
      Author Name: 
    </span>
    <span
      className="book__property__value"
    >
      mockName
       
    </span>
  </div>
  <div
    className="book__property"
  >
    <span
      className="book__property__name"
    >
      Author Gender: 
    </span>
    <span
      className="book__property__value"
    >
      mockGenre
       
    </span>
  </div>
</div>
`;
